From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Aditya Prayoga <aprayoga@users.noreply.github.com>
Date: Wed, 31 Jul 2019 12:51:00 +0200
Subject: [ARCHEOLOGY] Move mvebu DEFAULT, NEXT and DEV branch to next kernel
 (LTS) and U-boot #1426 (#1487)

> X-Git-Archeology: > recovered message: > * Initial Mvebu RFC https://github.com/armbian/build/issues/1426
> X-Git-Archeology: > recovered message: > Signed-off-by: Igor Pecovnik <igor.pecovnik@gmail.com>
> X-Git-Archeology: > recovered message: > * mvebu: add missing patches
> X-Git-Archeology: > recovered message: > Signed-off-by: Igor Pecovnik <igor.pecovnik@gmail.com>
> X-Git-Archeology: > recovered message: > * mvebu: change making u-boot targets to standard way, adjust patches and config
> X-Git-Archeology: > recovered message: > Signed-off-by: Igor Pecovnik <igor.pecovnik@gmail.com>
> X-Git-Archeology: > recovered message: > * helios4: set default branch to use U-Boot 2018.11
> X-Git-Archeology: > recovered message: > Switch over to U-Boot 2018.11 that has been used for some time in next
> X-Git-Archeology: > recovered message: > branch.
> X-Git-Archeology: > recovered message: > Signed-off-by: Aditya Prayoga <aditya@kobol.io>
> X-Git-Archeology: > recovered message: > * mvebu: helios4: Enable DEV branch
> X-Git-Archeology: > recovered message: > Signed-off-by: Aditya Prayoga <aditya@kobol.io>
> X-Git-Archeology: > recovered message: > * u-boot: Add RTC support on Clearfog and Helios4
> X-Git-Archeology: > recovered message: > Added DM driver for mvebu RTC and enable it on Clearfog and Helios4
> X-Git-Archeology: > recovered message: > configuration.
> X-Git-Archeology: > recovered message: > Signed-off-by: Aditya Prayoga <aditya@kobol.io>
> X-Git-Archeology: > recovered message: > * add boot-marvell.cmd backward compatibility
> X-Git-Archeology: > recovered message: > The patches added missing variable that used on boot-marvell.cmd and
> X-Git-Archeology: > recovered message: > also adjust the some memory addresses to prevent crash due to usage of
> X-Git-Archeology: > recovered message: > fdt_high and initrd_high.
> X-Git-Archeology: > recovered message: > Signed-off-by: Aditya Prayoga <aditya@kobol.io>
> X-Git-Archeology: > recovered message: > * helios4: Added SPI NOR flash target
> X-Git-Archeology: > recovered message: > Build bootable SPI NOR flash image.
> X-Git-Archeology: > recovered message: > Change the boot order to USB -> SATA -> MMC
> X-Git-Archeology: > recovered message: > Signed-off-by: Aditya Prayoga <aditya@kobol.io>
> X-Git-Archeology: > recovered message: > * Restore SPI support on U-Boot 2019.04
> X-Git-Archeology: > recovered message: > * mvebu: kernel: Added Wake-On-GPIO and WoL support
> X-Git-Archeology: > recovered message: > The patch set was missing during transition.
> X-Git-Archeology: > recovered message: > Signed-off-by: Aditya Prayoga <aditya@kobol.io>
> X-Git-Archeology: > recovered message: > * [#1429] SolidRun's ARMADA A388 SOM U-Boot ODT Update
> X-Git-Archeology: > recovered message: > Old versions of U-Boot did not configure correctly the ODT on data
> X-Git-Archeology: > recovered message: > signals of DDR RAM on SolidRun's ARMADA A388 SOMs.
> X-Git-Archeology: > recovered message: > According to SolidRun Knowledge Base, the changes already pushed to
> X-Git-Archeology: > recovered message: > mainline U-Boot. But then it was overwritten when Marvell DDR Training
> X-Git-Archeology: > recovered message: > Tool updated
> X-Git-Archeology: > recovered message: > [URL]
> X-Git-Archeology: > recovered message: > https://developer.solid-run.com/knowledge-base/armada-38x-som-u-boot-odt-update/
> X-Git-Archeology: > recovered message: > Signed-off-by: Aditya Prayoga <aditya@kobol.io>
> X-Git-Archeology: > recovered message: > * [#1429] mvebu: u-boot: Add revision id for Armada 38x B0
> X-Git-Archeology: > recovered message: > Added patch for SolidRun U-Boot v2018.01 and
> X-Git-Archeology: > recovered message: > for Helios4 U-Boot v2018.11
> X-Git-Archeology: > recovered message: > Signed-off-by: Aditya Prayoga <aditya@kobol.io>
> X-Git-Archeology: > recovered message: > * clearfog: Added SPI NOR flash target
> X-Git-Archeology: > recovered message: > Build bootable SPI NOR flash image.
> X-Git-Archeology: > recovered message: > Signed-off-by: Aditya Prayoga <aditya@kobol.io>
> X-Git-Archeology: > recovered message: > * mvebu: clearfog: DEV branch use mainline U-Boot
> X-Git-Archeology: > recovered message: > Also move clearfog base patch into its own board folder.
> X-Git-Archeology: > recovered message: > Signed-off-by: Aditya Prayoga <aditya@kobol.io>
> X-Git-Archeology: > recovered message: > * mvebu: enable U-Boot uart target
> X-Git-Archeology: > recovered message: > Normal MMC image can be used for uart boot using following command:
> X-Git-Archeology: > recovered message: > ./tools/kwboot -b u-boot-spl.kwb /dev/ttyUSBX
> X-Git-Archeology: > recovered message: > But on Helios4, the SPL failed to continue the booting process if ECC is
> X-Git-Archeology: > recovered message: > enabled, so disable it.
> X-Git-Archeology: > recovered message: > Since the usage of uart boot is more for rescue/debug, disable autoboot.
> X-Git-Archeology: > recovered message: > Signed-off-by: Aditya Prayoga <aditya@kobol.io>
> X-Git-Archeology: > recovered message: > * mvebu: NEXT branch use mainline U-Boot
> X-Git-Archeology: > recovered message: > Signed-off-by: Aditya Prayoga <aditya@kobol.io>
> X-Git-Archeology: > recovered message: > * mvebu: helios4: fix fancontrol related bug during buster testing
> X-Git-Archeology: > recovered message: > - On kernel 4.19, cpu thermal sensor changed the name from
> X-Git-Archeology: > recovered message: > armada_thermal into f10e4078.thermal. Added this new name to udev rules
> X-Git-Archeology: > recovered message: > - Since DEFAULT branch now use kernel 4.14, update fancontrol
> X-Git-Archeology: > recovered message: > configuration
> X-Git-Archeology: > recovered message: > - Load lm75 kernel module
> X-Git-Archeology: > recovered message: > - On kernel 4.19, cpu temp reading about 20 degree C lower, update
> X-Git-Archeology: > recovered message: > fancontrol configuration.
> X-Git-Archeology: > recovered message: > [URL]
> X-Git-Archeology: > recovered message: > https://forum.armbian.com/topic/10214-clearfogpro-possible-change-in-temperature-reporting-between-414next-and-419dev
> X-Git-Archeology: > recovered message: > Signed-off-by: Aditya Prayoga <aditya@kobol.io>
> X-Git-Archeology: > recovered message: > * mvebu: helios4: Override vendor provided fancontrol unit
> X-Git-Archeology: > recovered message: > systemd emit following message on dmesg
> X-Git-Archeology: > recovered message: > systemd[1]: /lib/systemd/system/fancontrol.service:9: PIDFile=
> X-Git-Archeology: > recovered message: > references path below legacy directory /var/run/, updating
> X-Git-Archeology: > recovered message: > /var/run/fancontrol.pid \xe2\x86\x92 /run/fancontrol.pid; please update
> X-Git-Archeology: > recovered message: > the unit file accordingly.
> X-Git-Archeology: > recovered message: > Override and change the value in the unit file to remove the message.
> X-Git-Archeology: > recovered message: > Signed-off-by: Aditya Prayoga <aditya@kobol.io>
> X-Git-Archeology: > recovered message: > * mvebu: DEV branch use its own u-boot patch folder
> X-Git-Archeology: > recovered message: > The patches are copied over from u-boot-mvebu-next
> X-Git-Archeology: > recovered message: > Signed-off-by: Aditya Prayoga <aditya@kobol.io>
> X-Git-Archeology: > recovered message: > * mvebu: u-boot: Make clearfog model distinction more obvious
> X-Git-Archeology: > recovered message: > While at it, also change SerDes LANE4 into USB 3.0 on Clearfog Base.
> X-Git-Archeology: > recovered message: > Signed-off-by: Aditya Prayoga <aditya@kobol.io>
> X-Git-Archeology: > recovered message: > * lib: Use apt-get install instead of dpkg on install_deb_chroot()
> X-Git-Archeology: > recovered message: > dpkg -i does not install dependencies required by the package.
> X-Git-Archeology: > recovered message: > This is needed if the BSP package requires other package that is not
> X-Git-Archeology: > recovered message: > installed during debootstrap.
> X-Git-Archeology: > recovered message: > Signed-off-by: Aditya Prayoga <aditya@kobol.io>
> X-Git-Archeology: > recovered message: > * config: mvebu: helios4: Move various tweak to family_tweak_bsp()
> X-Git-Archeology: > recovered message: > Various tweak in family_tweaks_s() only applied to SD card image.
> X-Git-Archeology: > recovered message: > Move it to family_tweaks_bsp() so it will also included on the BSP
> X-Git-Archeology: > recovered message: > package and applied to existing user.
> X-Git-Archeology: > recovered message: > Signed-off-by: Aditya Prayoga <aditya@kobol.io>
> X-Git-Archeology: > recovered message: > * config: mvebu: helios4: Add /etc/modules to BSP
> X-Git-Archeology: > recovered message: > On kernel 4.19, user need to modify the /etc/modules to add lm75 kernel
> X-Git-Archeology: > recovered message: > module. Pack the file into BSP so user no longer needed to modify it.
> X-Git-Archeology: > recovered message: > Signed-off-by: Aditya Prayoga <aditya@kobol.io>
> X-Git-Archeology: > recovered message: > * mvebu: kernel: Make zbud as built-in module
> X-Git-Archeology: > recovered message: > To remove the following error:
> X-Git-Archeology: > recovered message: > [    1.705485] zswap: default zpool zbud not available
> X-Git-Archeology: > recovered message: > [    1.705488] zswap: pool creation failed
> X-Git-Archeology: > recovered message: > Signed-off-by: Aditya Prayoga <aditya@kobol.io>
> X-Git-Archeology: > recovered message: > * bootscripts: mvebu: Add default value for spi_workaround
> X-Git-Archeology: > recovered message: > Signed-off-by: Aditya Prayoga <aditya@kobol.io>
> X-Git-Archeology: > recovered message: > * mvebu: kernel: Backport armada_thermal changes to 4.14 (#1452)
> X-Git-Archeology: > recovered message: > On kernel 4.19, armada_thermal driver has been fixed to address
> X-Git-Archeology: > recovered message: > Marvell's Errata #132698 (The changes first appear on LK 4.16). The
> X-Git-Archeology: > recovered message: > result is temperature reading is around 20 degree Celsius lower.
> X-Git-Archeology: > recovered message: > Currently armbian-motd apply -20C tweak for both LK 4.14 and LK 4.19
> X-Git-Archeology: > recovered message: > which is incorrect. Instead of adding some logic on what condition to
> X-Git-Archeology: > recovered message: > apply the tweak, it is better to remove the tweak and patch the kernel
> X-Git-Archeology: > recovered message: > instead.
> X-Git-Archeology: > recovered message: > Revert commit b3dd4e9 ("[ mvebu ] Put back Armada temperature tweak in
> X-Git-Archeology: > recovered message: > motd")
> X-Git-Archeology: > recovered message: > which is part of #1421 solution.
> X-Git-Archeology: > recovered message: > [URL]
> X-Git-Archeology: > recovered message: > https://forum.armbian.com/topic/10214-clearfogpro-possible-change-in-temperature-reporting-between-414next-and-419dev/
> X-Git-Archeology: > recovered message: > Signed-off-by: Aditya Prayoga <aditya@kobol.io>
> X-Git-Archeology: > recovered message: > * mvebu: helios4: unified fancontrol config
> X-Git-Archeology: > recovered message: > Since LK 4.14 on DEFAULT branch already patched and the temp reading is
> X-Git-Archeology: > recovered message: > same as LK 4.19 on NEXT branch, it is no longer needed to separate
> X-Git-Archeology: > recovered message: > fancontrol configuration file.
> X-Git-Archeology: > recovered message: > Signed-off-by: Aditya Prayoga <aditya@kobol.io>
> X-Git-Archeology: > recovered message: > * u-boot: helios4: Remove rev id patch
> X-Git-Archeology: > recovered message: > The patch is already applied in helios4 repo, no need to have it in
> X-Git-Archeology: > recovered message: > armbian.
> X-Git-Archeology: > recovered message: > This revert helios4 part of commit 7411c55
> X-Git-Archeology: > recovered message: > Signed-off-by: Aditya Prayoga <aditya@kobol.io>
> X-Git-Archeology: > recovered message: > * u-boot: clearfog: enable PCIe support and PCIe reset
> X-Git-Archeology: > recovered message: > Signed-off-by: Aditya Prayoga <aditya@kobol.io>
> X-Git-Archeology: > recovered message: > * u-boot: clearfog: add boot-marvell.cmd backward compatibility
> X-Git-Archeology: > recovered message: > The patches added missing variable that used on boot-marvell.cmd and
> X-Git-Archeology: > recovered message: > also adjust the some memory addresses to prevent crash due to usage of
> X-Git-Archeology: > recovered message: > fdt_high and initrd_high.
> X-Git-Archeology: > recovered message: > Signed-off-by: Aditya Prayoga <aditya@kobol.io>
> X-Git-Archeology: > recovered message: > * mvebu: helios4: tweak regarding temperature setting
> X-Git-Archeology: > recovered message: > Make fan speed similar compared to pre-patched armada-thermal. Target
> X-Git-Archeology: > recovered message: > PWM value around 70 during idle.
> X-Git-Archeology: > recovered message: > Signed-off-by: Aditya Prayoga <aditya@kobol.io>
> X-Git-Archeology: > recovered message: > * config: sources: clearfog to use u-boot 2018.01 for NEXT branch
> X-Git-Archeology: > recovered message: > This changes also affect Helios4. Moved the shared U-Boot source setting
> X-Git-Archeology: > recovered message: > back to Helios4 for NEXT branch.
> X-Git-Archeology: > recovered message: > Signed-off-by: Aditya Prayoga <aditya@kobol.io>
> X-Git-Archeology: > recovered message: > * config: boards: build Stretch image for Clearfog and Helios4
> X-Git-Archeology: > recovered message: > Also remove DEV from Helios4 CLI_TARGET
> X-Git-Archeology: > recovered message: > Signed-off-by: Aditya Prayoga <aditya@kobol.io>
> X-Git-Archeology: > recovered message: > * kernel: mvebu-next: Disable access to SPI Flash
> X-Git-Archeology: > recovered message: > User need to set spi_workaround=yes to enable SPI Flash access and lost
> X-Git-Archeology: > recovered message: > access to internal SATA.
> X-Git-Archeology: > recovered message: > Signed-off-by: Aditya Prayoga <aditya@kobol.io>
> X-Git-Archeology: > recovered message: > * mvebu-next: adjust kernel config
> X-Git-Archeology: > recovered message: > * mvebu-dev: bump to 5.2 and adjust kernel configuraion. Tested for building.
> X-Git-Archeology: > recovered message: > * Adjust kernel config, add AUFS
> X-Git-Archeology: > recovered message: > Signed-off-by: Igor Pecovnik <igor.pecovnik@gmail.com>
> X-Git-Archeology: > recovered message: > * mvebu-next: Adjust kernel config, add debug GPIO
> X-Git-Archeology: > recovered message: > Signed-off-by: Aditya Prayoga <aditya@kobol.io>
> X-Git-Archeology: > recovered message: > * mvebu-dev: separate Clearfog Base U-boot configuration file and patch
> X-Git-Archeology: > recovered message: > Signed-off-by: Aditya Prayoga <aditya@kobol.io>
> X-Git-Archeology: - Revision e261c6f82835bd9b12e07ba837b55fbf1aaa4327: https://github.com/armbian/build/commit/e261c6f82835bd9b12e07ba837b55fbf1aaa4327
> X-Git-Archeology:   Date: Wed, 31 Jul 2019 12:51:00 +0200
> X-Git-Archeology:   From: Aditya Prayoga <aprayoga@users.noreply.github.com>
> X-Git-Archeology:   Subject: Move mvebu DEFAULT, NEXT and DEV branch to next kernel (LTS) and U-boot #1426 (#1487)
> X-Git-Archeology:
> X-Git-Archeology: - Revision 150ac0c2afa147d9e3b036c8ecd8238fe5648cf3: https://github.com/armbian/build/commit/150ac0c2afa147d9e3b036c8ecd8238fe5648cf3
> X-Git-Archeology:   Date: Tue, 19 Nov 2019 23:25:39 +0100
> X-Git-Archeology:   From: Igor Pecovnik <igorpecovnik@users.noreply.github.com>
> X-Git-Archeology:   Subject: Remove K<4, change branches, new features (#1586)
> X-Git-Archeology:
> X-Git-Archeology: - Revision 2fe5e6f7553174c2b2be69d70398145e11af00b6: https://github.com/armbian/build/commit/2fe5e6f7553174c2b2be69d70398145e11af00b6
> X-Git-Archeology:   Date: Thu, 03 Dec 2020 10:24:27 +0100
> X-Git-Archeology:   From: Jannis <52237708+heisath@users.noreply.github.com>
> X-Git-Archeology:   Subject: [AR-558] Switch mvebu current to LK5.9 (based on previous mvebu-dev) (#2405)
> X-Git-Archeology:
> X-Git-Archeology: - Revision 0cdffb29b07305209efb12cf3b5ac6032d3a1153: https://github.com/armbian/build/commit/0cdffb29b07305209efb12cf3b5ac6032d3a1153
> X-Git-Archeology:   Date: Wed, 24 Mar 2021 19:01:53 +0100
> X-Git-Archeology:   From: Igor Pecovnik <igorpecovnik@users.noreply.github.com>
> X-Git-Archeology:   Subject: Renaming DEV branch to EDGE (#2704)
> X-Git-Archeology:
> X-Git-Archeology: - Revision b619876210d23cc34652b4b4fa15cf7f61703079: https://github.com/armbian/build/commit/b619876210d23cc34652b4b4fa15cf7f61703079
> X-Git-Archeology:   Date: Wed, 04 Jan 2023 18:37:42 +0100
> X-Git-Archeology:   From: Jannis <52237708+Heisath@users.noreply.github.com>
> X-Git-Archeology:   Subject: [AR-1313] Move mvebu edge to 6.1 (#4652)
> X-Git-Archeology:
> X-Git-Archeology: - Revision ff8412d676dc175edaf07f3cd64e47b428f1c3b6: https://github.com/armbian/build/commit/ff8412d676dc175edaf07f3cd64e47b428f1c3b6
> X-Git-Archeology:   Date: Mon, 05 Aug 2024 09:38:22 +0200
> X-Git-Archeology:   From: Igor Pecovnik <igor@armbian.com>
> X-Git-Archeology:   Subject: Add patches to 6.10 folder
> X-Git-Archeology:
> X-Git-Archeology: - Revision 9ea9e5b97b0fb5dd87727c64519084ef4cc26ec3: https://github.com/armbian/build/commit/9ea9e5b97b0fb5dd87727c64519084ef4cc26ec3
> X-Git-Archeology:   Date: Mon, 05 Aug 2024 09:38:22 +0200
> X-Git-Archeology:   From: Heisath <jannis@imserv.org>
> X-Git-Archeology:   Subject: Move to 6.8
> X-Git-Archeology:
> X-Git-Archeology: - Revision 72ce5c3f0b595438b71abfd5df194e9a7f0ea3d7: https://github.com/armbian/build/commit/72ce5c3f0b595438b71abfd5df194e9a7f0ea3d7
> X-Git-Archeology:   Date: Mon, 05 Aug 2024 09:38:22 +0200
> X-Git-Archeology:   From: Heisath <jannis@imserv.org>
> X-Git-Archeology:   Subject: Add mvebu 6.7 patches, move to rmk tree
> X-Git-Archeology:
---
 arch/arm/boot/dts/marvell/armada-38x-solidrun-microsom.dtsi | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/arm/boot/dts/marvell/armada-38x-solidrun-microsom.dtsi b/arch/arm/boot/dts/marvell/armada-38x-solidrun-microsom.dtsi
index 111111111111..222222222222 100644
--- a/arch/arm/boot/dts/marvell/armada-38x-solidrun-microsom.dtsi
+++ b/arch/arm/boot/dts/marvell/armada-38x-solidrun-microsom.dtsi
@@ -107,6 +107,7 @@ w25q32: flash@0 {
 		compatible = "w25q32", "jedec,spi-nor";
 		reg = <0>; /* Chip select 0 */
 		spi-max-frequency = <3000000>;
+		status = "disabled";
 	};
 };
 
-- 
Armbian

